src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
> This is just a copy-paste from isc_file.cpp::IConv class.
This is not true. 
I see "class IConv" in isc_file.cpp (yes, bad name too) while you introduced 3 new classes and none of them is "IConv".
I.e. you already made changes in copied code.
If you don't want to do something - say "i don't want", no need to create empty reasoning...

> I leave it to one who will finish this POSIX port.
So, you ask us to merge half-made code and to "finish" it after you ? Cool...